src/coreclr/jit/importer.cpp

```diff
@@ -14997,53 +14997,60 @@ void Compiler::impImportBlockCode(BasicBlock* block)
                 tiRetVal        = se.seTypeInfo;
                 op1             = tree;
 
-                // If the expression to dup is simple, just clone it.
-                // Otherwise spill it to a temp, and reload the temp twice.
-                bool cloneExpr = false;
+                // In unoptimized code we leave the decision of
+                // cloning/creating temps up to impCloneExpr, while in
+                // optimized code we prefer temps except for some cases we know
+                // are profitable.
 
-                if (!opts.compDbgCode)
+                if (opts.OptimizationEnabled())
                 {
+                    bool clone = false;
                     // Duplicate 0 and +0.0
                     if (op1->IsIntegralConst(0) || op1->IsFloatPositiveZero())
                     {
-                        cloneExpr = true;
+                        clone = true;
                     }
                     // Duplicate locals and addresses of them
                     else if (op1->IsLocal())
                     {
-                        cloneExpr = true;
+                        clone = true;
                     }
-                    else if (op1->TypeIs(TYP_BYREF) && op1->OperIs(GT_ADDR) && op1->gtGetOp1()->IsLocal() &&
+                    else if (op1->TypeIs(TYP_BYREF, TYP_I_IMPL) && impIsAddressInLocal(op1) &&
                              (OPCODE)impGetNonPrefixOpcode(codeAddr + sz, codeEndp) != CEE_INITOBJ)
                     {
-                        cloneExpr = true;
+                        // We mark implicit byrefs with GTF_GLOB_REF (see gtNewFieldRef for why).
+                        // Avoid cloning for these.
+                        clone = (op1->gtFlags & GTF_GLOB_REF) == 0;
```

Comment:
It is a CQ problem?

Side note: I suspect the conservative `GLOB_REF` setting in `gtNewFieldRef` is no longer necessary now that we always mark the expanded indirection with it in morph.

Reply:
I think this was just working around below assert. IIRC there were no large diffs associated with this case.

Getting rid of the special case in `gtNewFieldRef` would be nice. I tried in #66266 but had to revert it (#66415). If morph is conservative about marking these now it is probably doable again.

Reply:
> I think this was just being conservative about below assert. IIRC there were no large diffs associated with this case.

I see. Another thing to consider here is that addresses of struct field chains (i. e. `ADDR(FIELD(ADDR(FIELD(...))))` are marked with `GLOB_REF`, so it's not just implicit byrefs that are being excluded.

Reply:
I opened #74563 for it.